templates/addons/metrics-server/kustomization.yaml

```diff
@@ -0,0 +1,15 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+namespace: kube-system
+resources:
+  - https://github.com/kubernetes-sigs/metrics-server/releases/download/v0.5.2/components.yaml
```

Comment:
this is awesome, we should use this model for more stuff (eg. cloud-provider) instead of maintaining addon specs in the repo

Reply:
cool!  I created a tracking issue with a few possible items: https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/1917